### PR TITLE
Don't build examples when running unit tests

### LINKED ISSUE #3117: Examples being built when running tests
### Introduce the problem
The examples are being built when running tests, is this necessary? (also the demos are being built)

### Examples and cases
Run the test cases and see the examples being built.

### Suggested solution
Don't build examples when unit tests are being build.

### PR DESCRIPTION
**Pull request by C47D**
### Description of the feature or fix

@cmumford @embeddedt I was trying to get the examples not being compiled when running the unit tests, I'm not familiar with CMakeLists.txt so I thought by setting `LV_BUILD_EXAMPLES=0` and removing the `target_link_libraries` when compiling the test would be enough, but it seems like it doesn't, maybe I'm missing something?

Purpose of the change is to run tests faster, I'm planning to try run tests individually or passing a list of tests cases to run to the python script. 

Closes #3117 

### Checkpoints
- [ ] Follow the [styling guide](https://github.com/lvgl/lvgl/blob/master/docs/CODING_STYLE.md)
- [ ] Run `code-format.py` from the `scripts` folder. [astyle](http://astyle.sourceforge.net/install.html) needs to be installed.
- [ ] Update the documentation

### CODE REVIEW

File: tests/CMakeLists.txt [hunk 1]
```diff
@@ -97,7 +97,7 @@ set(LVGL_TEST_OPTIONS_16BIT
 
 set(LVGL_TEST_OPTIONS_16BIT_SWAP
     -DLV_COLOR_DEPTH=16
-    -DLV_COLOR_16_SWAP=1
```
Comment (cmumford) [inline tests/CMakeLists.txt:100]: I think it would be cleaner if the whitespace fixes (which are good) landed in a separate commit titled something like "white fixes". This makes `git blame` a bit more helpful.

File: tests/CMakeLists.txt [hunk 2]
```diff
@@ -328,6 +330,7 @@ add_library(test_common
         src/test_fonts/font_3.c
         unity/unity_support.c
         unity/unity.c
+        src/img_caret_down.c
```
Comment (cmumford) [inline tests/CMakeLists.txt]: Did you copy `examples/assets/img_caret_down.c` into `tests/src` and forget to add it to this commit? An alternative is to change this to:

    ${LVGL_DIR}/examples/assets/img_caret_down.c

I'd be in favor of moving the example asset files into their own library on which examples and tests both depend, but that's outside the scope of this change. @embeddedt WDYT about this?

File: tests/CMakeLists.txt (hunk 1 above)
Comment (C47D) [reply tests/CMakeLists.txt:100]: I agree, how can I fix it? Reverting this commit and then making two separate commits?

File: tests/CMakeLists.txt (hunk 2 above)
Comment (C47D) [reply tests/CMakeLists.txt]: Agree, I can try to update the path instead.

File: tests/CMakeLists.txt (hunk 1 above)
Comment (cmumford) [reply tests/CMakeLists.txt:100]: I think the easiest is for you to make a second commit with the whitespace changes reverted. The reviewer, when approving your PR, can use [squash and merge](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/incorporating-changes-from-a-pull-request/about-pull-request-merges#squash-and-merge-your-pull-request-commits). Alternatively you could rebase/squash and then do a force push to your branch.

File: tests/CMakeLists.txt (hunk 2 above)
Comment (cmumford) [reply tests/CMakeLists.txt]: Actually I tried that a little bit later and because your change, by design, doesn't define `LV_BUILD_EXAMPLES` the `img_caret_down`  symbol is not defined. I wasn't testing correctly so I missed that :-(.

File: tests/CMakeLists.txt (hunk 1 above)
Comment (C47D) [reply tests/CMakeLists.txt:100]: Sorry, I thought I was sending the changes from #3133 into this commit.

"Problem" is I'm using a git hook that look for trailing whitespaces and fixes them in the staged files just before the commit, so if we want to prevent that I would need to skip that git hook, do the commit and then force the git hook to fix the trailing whitespaces. Or before any change force the git hook on the file I need to work on.

Comment (cmumford) [reply tests/CMakeLists.txt:100]: Maybe not worth it then. You could do the whitespace fixup in a separate commit and then rebase your branch on master once it lands, but that's up to you. I'm not a project collaborator with PR approve privileges, I'm just a contributor.  Maybe keep the whitespace in this PR and do that only if requested by the collaborator.

File: tests/CMakeLists.txt (hunk 2 above)
Comment (C47D) [reply tests/CMakeLists.txt]: Hi, thanks for your patience, the dropdown test file uses that particular asset file. I haven't worked on this since last week, will try to continue this week.

**Comment by C47D** [comment]
So the examples are no longer being build when running unit tests, they are being build when running the build tests. Still trying to figure out how to let the unit tests configuration where's the `img_caret_down` asset.

**Comment by kisvegabor** [comment]
I think the best would be to make the tests use their own assets. I.e. we can add a `test/src/test_assets` folder. What do you think?

**Comment by C47D** [comment]
It is ok for me, should the directories `ref_imgs` and `test_fonts` moved to that new directory?

**Comment by kisvegabor** [comment]
`ref_imgs` are not really assets so I'd keep them in separate folder. 

`test_fonts` actually contains `*.fnt` assets and `*.c` reference fonts. 

What about creating an `assets` and a `refs` folder too? (`*.fnt` would go to the `refs` folder)

**Comment by C47D** [comment]
> What about creating an assets and a refs folder too? (*.fnt would go to the refs folder)

Ok, I will do that

**Comment by C47D** [comment]
So this now builds, but when I tried to validate again the examples are not being build when running tests it seems like they are, so, could you test it in your end?

I was checking the `custom.cmake` file and it seems like it adds lv_examples as dependency for LVGL, I may be wrong about this tho.

**Comment by kisvegabor** [comment]
`./main.py test --clean` builds the examples for me too.

**Comment by C47D** [comment]
Will upload a diff I did on custom.cmake later today
